### PR TITLE
Port over `manageExistingSafes` saga

### LINKED ISSUE #688: Port `manageExistingSafes` saga
Clarifications:
- Just like `managePermissions` or `manageReputation`, this saga should be capable of adding or removing safes. It's not needed that it can do both things at the same time as the add and remove safe dialogs are separated.
- This saga should use `editColony` to add/remove safes as we are storing them in the colony's metadata

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Test this by adding and removing safes through the flows of the dialogs.

- After adding a safe, the safe should be available for removal in the "Remove Safe" dialog.
- After removing a safe, it should no longer appear in the "Remove Safe" dialog list and you should be able to re-add it

If you are looking for safes to use, check this draft [here](https://github.com/orgs/JoinColony/projects/25?pane=issue&itemId=31538206).

Resolves #688

### CODE REVIEW

**Review by thomassdk — APPROVED** [review]
Works well for me just some minor comments vis a vis naming :)

File: src/components/common/Dialogs/AddExistingSafeDialog/AddExistingSafeDialog.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const AddExistingSafeDialog = ({
       >
         <AddExistingSafeDialogForm
           networkOptions={networkOptions}
-          colonySafes={[]} // safes
+          colonySafes={colony.metadata?.safes || []}
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/common/Dialogs/AddExistingSafeDialog/AddExistingSafeDialog.tsx:77]: Your hunch is right! I left this prop's name like that to make it clear that is an array with the safes that are already linked to the colony. However, after taking a closer look, this prop is no longer being used by the steps, therefore we can remove it altogether :)

File: src/redux/sagas/actions/manageExistingSafes.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    // const parsedColonyMetadata = JSON.parse(currentMetadata);
+    // const metadataVersion = getEventMetadataVersion(currentMetadata);
+    // const currentColonyMetadata =
+    //   metadataVersion === 1
+    //     ? { ...parsedColonyMetadata }
+    //     : parsedColonyMetadata.data;
+
+    let updatedSafes: Safe[];
+
+    const currentColonySafes =
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/redux/sagas/actions/manageExistingSafes.ts:142]: For sure

File: src/redux/sagas/actions/manageExistingSafes.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    //     ? { ...parsedColonyMetadata }
+    //     : parsedColonyMetadata.data;
+
+    let updatedColonySafes: Safe[];
+
+    const colonySafes =
+      colony.metadata?.safes
+        ?.filter(notNull)
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        .map(({ __typename, ...rest }) => rest) || [];
```
Comment (jakubcolony) [inline src/redux/sagas/actions/manageExistingSafes.ts:146]: There's an `excludeTypenameKey` util in `~utils/object` that does that without needing to disable eslint rules

File: src/redux/types/actions/colonyActions.ts [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { ColonyRole } from '@colony/colony-js';
 
 import { ActionTypes } from '~redux';
 import { Address, WithKey, DomainColor, Domain, Colony } from '~types';
+import { Safe } from '~gql';
```
Comment (jakubcolony) [inline src/redux/types/actions/colonyActions.ts:7]: Shouldn't that be imported from `~types`?

**Review by jakubcolony — APPROVED** [review]
Looking good 👍

**Pull request merged**